lib/tasks/belnet.rake

```diff
@@ -0,0 +1,27 @@
+namespace :belnet do
+
+  namespace :users do
+
+      desc "Deletes users who never had any activity, after some time"
+      task :clean => :environment do 
+
+        User.where(last_sign_in_at:nil).find_each do |user|
+
+          # Only consider users that have been created more than 1 year ago
+          if user.created_at.year < ( Date.today.year - 1 )
+
+            # Unlink user from any existing plan
+            role = Role.where(user_id:user.id).destroy_all
```

Comment:
Normally this is not necessary, as the roles are cleaned up automatically.

See https://github.com/DMPbelgium/roadmap/blob/master/app/models/user.rb#L98 and look for [dependent](https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-dependent), as long as you do
not use `user.delete` which executes the raw query without any callbacks

Reply:
ok but at least it works :-)